xonsh/ptk/shortcuts.py

```diff
@@ -0,0 +1,451 @@
+"""A prompt-toolkit inspired shortcut collection."""
```

Comment:
It is mostly from prompt_toolkit stuff that can hopefully be removed relatively soon.  I wouldn't mind passing on this since pretty much the major offenders in the bottom two methods (300+ lines) can all go away on the next prompt-toolkit release.